### ISSUE TITLE
CSS custom properties as component props not recognized

### ISSUE THREAD

**Issue by janosh** [pheuter/svelte-check-rs#67]
`svelte-check-rs` reports false positive errors for [CSS custom properties passed as component props](https://svelte.dev/docs/svelte/component-fundamentals#style-props), which is valid Svelte 5 syntax.

## Reproduction

Create two files:

**`Box.svelte`**
```svelte
<script lang="ts">
  let { color = `red` }: { color?: string } = $props()
</script>

<div style:background={color}>
  <slot />
</div>
```

**`App.svelte`**
```svelte
<script lang="ts">
  import Box from './Box.svelte'
</script>

<Box --some-css-var="blue">
  content
</Box>
```

## Error

```
Error: Object literal may only specify known properties, and '"--some-css-var"' does not exist in type '__SvelteProps_Box_'. (ts(TS2353))
```

## Expected

No error. CSS custom property props (`--*`) should be allowed on any component.
